### PR TITLE
Add identical hostnames tests for desired behavior

### PR DESCRIPTION
**Pull request by trepel**
## Overview

Last piece to address https://github.com/Kuadrant/testsuite/issues/406

Follows up on https://github.com/Kuadrant/testsuite/pull/439 and https://github.com/Kuadrant/testsuite/pull/462 which added similar tests for AuthPolicy and  RateLimitPolicy (RLP) respectively, but these tests validating current behavior.

The tests in this PR cover two scenarios, both testing desired behavior. Desired behavior means that Kuadrant does not behave like that yet so the tests are expected to fail. Once they start passing it means that implementation has changed and it would be good to take a closer look how and if that makes sense and update the testsuite accordingly.

### Scenario 1.
Test that 2nd Gateway-attached AuthPolicy is actually enforced on route (currently it states that it is enforced but effectively it is ignored)

### Scenario 2.
Test that 2nd HTTPRoute-attached AuthPolicy is enforced (currently it fails with an error, status being Enforced=False)

### Scenario 3.
Test that the 1st HTTPRoute-attached RLP is effectively enforced (currently it is effectively ignored because the 2nd HTTPRoute-attached RLP "kicks" it off)

## Verification Steps

Run the three tests. Or even better run the whole dir with all the "identical hostname" tests. 4 tests should pass, 3 should be xfailed. It should take roughly 8-9 minutes:

```
$ make testsuite/tests/kuadrant/identical_hostnames/
...
======= 4 passed, 2 xfailed in 473.34s (0:07:53) =======
```

### CODE REVIEW

File: testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_rlp_on_gw_and_route.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def rate_limit2(request, gateway, blame, cluster, label):
     return rlp
 
 
-def test_identical_hostnames_rlp_on_gw_and_route_ignored(client, rate_limit, rate_limit2, hostname):
```
Comment (pehala) [inline testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_rlp_on_gw_and_route.py:25]: Could you extract these unrelated changes to a separate PR?

File: testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_auth_on_gw_and_route_desired.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    auth_policy.authorization.add_opa_policy("rego", "allow = false")
+    request.addfinalizer(auth_policy.delete)
+    auth_policy.commit()
+    auth_policy.wait_for_ready()
+    return auth_policy
+
+
+@pytest.mark.xfail(
+    strict=True,
+    reason="Currently the Gateway-attached Policy is ignored so 200 (OK) is returned instead of 403 (Forbidden)",
```
Comment (pehala) [inline testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_auth_on_gw_and_route_desired.py:28]: I am missing issue reference, could you add pytest.mark.issue as well?

File: testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_auth_on_gw_and_route_desired.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    auth_policy = AuthPolicy.create_instance(openshift, blame("authz2"), gateway, labels={"testRun": label})
+    auth_policy.authorization.add_opa_policy("rego", "allow = false")
+    request.addfinalizer(auth_policy.delete)
+    auth_policy.commit()
+    auth_policy.wait_for_ready()
+    return auth_policy
+
+
+@pytest.mark.xfail(
+    strict=True,
```
Comment (pehala) [inline testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_auth_on_gw_and_route_desired.py:27]: I think this is not needed, we have xpassfail enabled by default

File: testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_rlp_on_gw_and_route.py (hunk 1 above)
Comment (trepel) [reply testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_rlp_on_gw_and_route.py:25]: Sure, will do

File: testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_auth_on_gw_and_route_desired.py (hunk 2 above)
Comment (trepel) [reply testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_auth_on_gw_and_route_desired.py:28]: I don't think that there is one. Currently this is only documented as a limitation.

File: testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_auth_on_gw_and_route_desired.py (hunk 3 above)
Comment (trepel) [reply testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_auth_on_gw_and_route_desired.py:27]: OK, will remove this for all xfalis across identical hostname tests

File: testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_auth_on_gw_and_route_desired.py (hunk 2 above)
Comment (pehala) [reply testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_auth_on_gw_and_route_desired.py:28]: I would create it or link epic where this is mentioned (I think there was one)

Comment (trepel) [reply testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_auth_on_gw_and_route_desired.py:28]: D&O Epic mentions this: https://github.com/Kuadrant/kuadrant-operator/issues/431 (Tier 2).

File: testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_auth_on_gw_and_route_desired.py (hunk 3 above)
Comment (trepel) [reply testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_auth_on_gw_and_route_desired.py]: Done

File: testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_auth_on_gw_and_route_desired.py (hunk 2 above)
Comment (trepel) [reply testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_auth_on_gw_and_route_desired.py]: Done

File: testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_rlp_on_gw_and_route.py (hunk 1 above)
Comment (trepel) [reply testsuite/tests/kuadrant/identical_hostnames/test_identical_hostnames_rlp_on_gw_and_route.py]: Done, removed from this PR. Created https://github.com/Kuadrant/testsuite/pull/479 instead

**Review by pehala — DISMISSED** [review]

**Comment by pehala** [comment]
Needs rebase due to #473 (sorry :D)

**Comment by trepel** [comment]
> Needs rebase due to #473 (sorry :D)

Grrrr! :) Rebased!

**Review by pehala — APPROVED** [review]

**Review by averevki — APPROVED** [review]
